File: go/count-items-matching-a-rule.go
func GetKeyIndex(ruleKey string) int {
    switch ruleKey {
        case "type": return 0
        case "color": return 1
        case "name": return 2
        default: return -1
    }
}

func countMatches(items [][]string, ruleKey string, ruleValue string) int {
    index := GetKeyIndex(ruleKey)
    matches := 0
    for i := 0; i < len(items); i++ {
        if items[i][index] == ruleValue {
            matches += 1
        }
    }
    return matches
}

File: python/count-items-matching-a-rule.py
KEY_MAP = {
    "type": 0,
    "color": 1,
    "name": 2,
}

class Solution:
    def countMatches(self, items: List[List[str]], ruleKey: str, ruleValue: str) -> int:
        index = KEY_MAP[ruleKey]
        matches = 0
        for item in items:
            if item[index] == ruleValue:
                matches += 1
        return matches
